### PR TITLE
Clustering changes for gitrest and historian

### PR DESCRIPTION
**Pull request by Toritos01**
## Description
This is a followup PR to: https://github.com/microsoft/FluidFramework/pull/19551
It takes the changes of that previous PR and also applies them to gitrest and historian (previously it was only on r11s).
The changes allow for the support of Redis instances with clustering enabled.

### CODE REVIEW

File: server/historian/packages/historian-base/package.json [hunk 1]
```diff
@@ -55,12 +55,12 @@
 		"@fluidframework/common-utils": "^1.1.1",
 		"@fluidframework/gitresources": "^3.0.0",
 		"@fluidframework/protocol-definitions": "^1.1.0",
-		"@fluidframework/server-services": "^3.0.0",
-		"@fluidframework/server-services-client": "^3.0.0",
-		"@fluidframework/server-services-core": "^3.0.0",
-		"@fluidframework/server-services-shared": "^3.0.0",
-		"@fluidframework/server-services-telemetry": "^3.0.0",
-		"@fluidframework/server-services-utils": "^3.0.0",
+		"@fluidframework/server-services": "3.1.0-236733",
```
Comment (znewton) [inline server/historian/packages/historian-base/package.json:58]: Let's use `^3.1.0-236733` for version range

Comment (Toritos01) [reply server/historian/packages/historian-base/package.json:58]: Fixed 😄

File: server/gitrest/packages/gitrest-base/src/utils/filesystems.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@
  */
 
 import fs from "node:fs";
-import { Redis as IoRedis, RedisOptions as IoRedisOptions } from "ioredis";
+import { Redis as IoRedis, RedisOptions as IoRedisOptions, type Cluster } from "ioredis";
```
Comment (zhangxin511) [inline server/gitrest/packages/gitrest-base/src/utils/filesystems.ts:7]: Educate me, why there is a `type` here but others not?

**Review by zhangxin511 — APPROVED** [review]

File: server/gitrest/packages/gitrest-base/src/utils/filesystems.ts (hunk 2 above)
Comment (Toritos01) [reply server/gitrest/packages/gitrest-base/src/utils/filesystems.ts:7]: For this one, I think I was getting a compiler error, with the recommended change being to add the "type".
But not sure why this is needed here and not in the others.

**Pull request merged**